webauthn-rs-proto/src/options.rs

```diff
@@ -197,6 +197,20 @@ pub enum AuthenticatorAttachment {
     CrossPlatform,
 }
 
+/// The Relying Party's requirements for client-side discoverable credentials.
+///
+/// <https://www.w3.org/TR/webauthn-2/#enumdef-residentkeyrequirement>
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
+#[serde(rename_all = "lowercase")]
+pub enum ResidentKeyRequirement {
+    /// <https://www.w3.org/TR/webauthn-2/#dom-residentkeyrequirement-discouraged>
+    Discouraged,
+    /// <https://www.w3.org/TR/webauthn-2/#dom-residentkeyrequirement-preferred>
+    Preferred,
```

Comment:
Please add a comments warning like
"⚠️ In all major browsers preferred is identical in behaviour to required. You should use required instead."